### PR TITLE
Aliasing math expression shows error earlier

### PR DESCRIPTION
**Pull request by kubouch**
# Description

Previously, if attempting to alias a non-call expression, it would result in strange error messages when trying to use the alias. Now, if we're aliasing a math expression, we throw an error early.

Before:
```
> alias spam = 'foo'

> spam
Error: nu::shell::external_command

  × External command failed
   ╭─[entry #4:1:1]
 1 │ spam
   · ──┬─
   ·   ╰── did you mean 'foo'?
   ╰────
  help: No such file or directory (os error 2)
```

After:
```
> alias spam = 'foo'
Error: nu::parser::cant_alias_expression

  × Can't create alias to expression.
   ╭─[entry #1:1:1]
 1 │ alias spam = 'foo'
   ·              ──┬──
   ·                ╰── aliasing String is not supported
   ╰────
  help: Only command calls can be aliased.
```

Fixes https://github.com/nushell/nushell/issues/8760
Fixes https://github.com/nushell/nushell/issues/8249

# User-Facing Changes

Better error messages. Might break some existing code that previously failed silently.

# Tests + Formatting

Don't forget to add tests that cover your changes.

Make sure you've run and fixed any issues with these commands:

- `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- `cargo clippy --workspace -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- `cargo test --workspace` to check that all tests pass
- `cargo run -- crates/nu-utils/standard_library/tests.nu` to run the tests for the standard library

> **Note**
> from `nushell` you can also use the `toolkit` as follows
> ```bash
> use toolkit.nu  # or use an `env_change` hook to activate it automatically
> toolkit check pr
> ```

# After Submitting

If your PR had any user-facing changes, update [the documentation](https://github.com/nushell/nushell.github.io) after the PR is merged, if necessary. This will help us keep the docs up to date.

### CODE REVIEW

File: crates/nu-parser/src/errors.rs [hunk 1]
```diff
@@ -109,6 +109,13 @@ pub enum ParseError {
     )]
     CantAliasKeyword(String, #[label("not supported in alias")] Span),
 
+    #[error("Can't create alias to expression.")]
+    #[diagnostic(
+        code(nu::parser::cant_alias_expression),
+        help("Only command calls can be aliased.")
+    )]
+    CantAliasExpression(String, #[label("aliasing {0} is not supported")] Span),
```
Comment (sholderbach) [inline crates/nu-parser/src/errors.rs:117]: Maybe worth suggesting using backticks to quote a complex command name?

Comment (kubouch) [reply crates/nu-parser/src/errors.rs:117]: It's not about command name. It's about trying to alias some other expression than a call, like a string, range, etc.

Comment (sholderbach) [reply crates/nu-parser/src/errors.rs:117]: Yeah but imagine you try to alias an external command with space in its absolute path like on windows, and the first thing you would possibly try is the string syntax using `"`

Comment (kubouch) [reply crates/nu-parser/src/errors.rs:117]: I think that one you won't be able to detect easily. Imagine a call `foo bar.txt`, you can't know whether you're supposed to call `foo` command with `bar.txt` argument, or a full `foo bar.txt` file. It parses as an external and will skip the math expression test as well, so I can't detect it.

Comment (kubouch) [reply crates/nu-parser/src/errors.rs:117]: I'm gonna merge this because it's blocking my other PR, but we can discuss and improve this further.

**Pull request merged**